File: beecrowd/python/Estruturas e Bibliotecas/1023.py
from operator import itemgetter
cidade = 1
while True:
    N = int(input())
    if N == 0:
        break
    pessoas = 0
    consumo_total = 0
    valores = {}
    for i in range(N):
        X, Y = map(int, input().split())
        pessoas += X
        consumo_total += Y
        consumo = Y//X
        if consumo in valores: 
            valores[consumo] += X
        else:
            valores[consumo] = X
    #print(valores)
    consumo_medio = consumo_total/pessoas
    dados_ordenados = dict(sorted(valores.items()))
    #print(dados_ordenados)
    print(f"Cidade# {cidade}:")
    for chave, valor in dados_ordenados.items():
        print(f"{valor} - {chave}", end=" ")
    print()
    print(f"Consumo medio: {consumo_medio:.2f} m3.")
    print()
    cidade += 1

File: beecrowd/c/Iniciante/Estuturas e Bibliotecas/1023.c
#include <stdio.h>

int main()
{
    int cidade = 1;
    int N;
    int consumo;
    int X, Y;
    while (scanf("%d", &N) && N != 0) {
        long long pessoas = 0;
        long long consumo_total = 0;  
        int valores[201] = {0};
        
        for(int i=0; i<N; i++){
            scanf("%d %d", &X, &Y);
            pessoas+=X;
            consumo_total+=Y;
            consumo = Y/X;
            valores[consumo] += X;
        }
        long long medio_inteiro = ((long long)consumo_total * 100) / pessoas;
        printf("Cidade# %d:\n", cidade);
        for(int c = 0; c <= 200; c++){
            if (valores[c]>0){
                printf("%d-%d ", valores[c], c);
            }
        }
        printf("\n");
        printf("Consumo medio: %lld.%02lld m3.\n", medio_inteiro / 100, medio_inteiro % 100);
        printf("\n");
        cidade++;
    }
    return 0;
}